listener.go

```diff
@@ -132,7 +146,7 @@ func (h *hub) processReqDequeue(req baps3.Message) (resps []*baps3.Message) {
 }
 
 func (h *hub) processReqEnqueue(req baps3.Message) (resps []*baps3.Message) {
-	args := req.AsSlice()[1:]
+	args := req.Args()
 	if len(args) != 4 {
 		return append(resps, baps3.NewMessage(baps3.RsWhat).AddArg("Bad command"))
```

Comment:
Could probably farm this out into a separate function, since it's fairly nontrivial and repeated?